### ISSUE TITLE
Incorrect parsing of example when discriminatorValue is used with union type

### ISSUE THREAD

**Issue by rmihael** [mulesoft/api-workbench#330]
```yaml
#%RAML 1.0
title: Example API

types:
  Base:
    type: object
    discriminator: type
    properties:
      type: string

  TypeA:
    type: Base
    discriminatorValue: type_a

  TypeB:
    type: Base
    discriminatorValue: type_b

  Optional:
    type: object
    properties:
      attr: Base?
    example:
      attr:
        type: type_a
```

The code example above fails to parse in my installation of API Workbench. This is the error I'm getting:

> Union type option does not pass validation (Base: None of the 'Base' type known subtypes declare 'type_a' as value of discriminating property 'type'.

Changing `attr: Base?` to `attr?: Base` solves the problem, but I need to express attribute that accepts null values.

API Workbench version is 0.8.38 and Atom.io version is 1.11.1

**Comment by ddenisenko** [comment]
Reproducible, thank you.

**Comment by KonstantinSviridov** [comment]
Will be fixed in 0.8.40
